parlai/crowdsourcing/projects/multisession_chat/human_eval/worlds.py

```diff
@@ -0,0 +1,173 @@
+        accordingly.
+        """
+
+        control_msg = {"episode_done": False}
+
+        time.sleep(2)
+        coordinator_first_msg = {
+            'episode_done': False,
+            'id': 'Coordinator',
+            'text': 'Please chitchat with another worker for 6 turns as if you were catching up since last time you two spoke.',
+            'fake_start': True,
+            'agent_idx': 2,
+            'task_data': self.task_data,
+        }
+        self.agent.observe(validate(coordinator_first_msg))
+        time.sleep(2)
+        human_first_msg = {
+            'episode_done': False,
+            'id': self.agent.id,
+            'text': self.context_for_bot_prompt,
+            'fake_start': True,
+            'agent_idx': 0,
+            'task_data': self.task_data,
+        }
+        for k, v in control_msg.items():
+            human_first_msg[k] = v
+
+        # self.dialog.append(human_first_msg)
+        # self.agent.observe(validate(human_first_msg))
+        print(human_first_msg)
+        self.bot.observe(validate(human_first_msg))
+
+        first_bot_act = self.bot.act()
+        first_bot_act = Compatibility.maybe_fix_act(first_bot_act)
+        first_bot_act['id'] = 'THEY'
+        self.agent.observe(validate(first_bot_act))
+
+        bot_utterance_data = {
+            'agent_idx': 1,
+            'text': first_bot_act['text'],
+            'id': 'THEY',
+        }
+        self.dialog.append(bot_utterance_data)
+
+    def get_final_chat_data(self) -> Dict[str, Any]:
+        """
+        Add non-image-chat-specific fields to the final chat data.
+        """
+        data = super().get_final_chat_data()
+        context_data = {
+            'model_name': self.model_name,
+            'context_info': self.context_info,
+            'bot_persona_strings': self.bot_persona_strings,
+            'human_persona_strings': self.human_persona_strings,
+            'initial_task_data': self.task_data,
+        }
+        data.update(context_data)
+        return data
+
+    def _prepare_acceptability_checking(self) -> Tuple[List[str], List[str]]:
+        """
+        Apply acceptability checking params specific to BST-style conversation.
+
+        The BST mode starts the conversation with two previous utterances, so there
+        should be no new greeting. Also, the first human response is one of the previous
+        utterances, so it shouldn't get checked.
+        """
+        human_messages = [
+            message['text'] for message in self.dialog if message['id'] == 'YOU'
+        ]
+        violation_types = ['min_words', 'all_caps', 'exact_match']
+        if self.opt['conversation_start_mode'] == 'blended_skill_talk':
+            violation_types.append('penalize_greetings')
+            # human_messages = human_messages[1:]
+        return human_messages, violation_types
+
+
+def make_world(opt, agents, initialization_data):
```

Comment:
Hmm can you add a TODO to one day merge this function in with the `make_world` in in `parlai/crowdsourcing/tasks/model_chat/worlds.py`? Functionally I think they do the same things